controllers/storagecluster/cephcluster_test.go

```diff
@@ -332,39 +374,39 @@ func TestStorageClassDeviceSetCreation(t *testing.T) {
-			assert.Equal(t, deviceSet.DataPVCTemplate, scds.VolumeClaimTemplates[0])
-			assert.Equal(t, true, scds.Portable)
-			assert.Equal(t, c.sc.Spec.Encryption.Enable, scds.Encrypted)
+			assert.Equal(t, fmt.Sprintf("%s-%d", deviceSet.Name, i), scds.Name, c.label)
+			assert.Equal(t, deviceSet.Count/3, scds.Count, c.label)
+			assert.Equal(t, defaults.DaemonResources["osd"], scds.Resources, c.label)
+			assert.Equal(t, deviceSet.DataPVCTemplate, scds.VolumeClaimTemplates[0], c.label)
+			assert.Equal(t, true, scds.Portable, c.label)
+			assert.Equal(t, c.sc.Spec.Encryption.Enable, scds.Encrypted, c.label)
 			if scds.Portable && c.topologyKey == "rack" {
-				assert.Equal(t, scds.Placement.TopologySpreadConstraints[0].WhenUnsatisfiable, corev1.UnsatisfiableConstraintAction("DoNotSchedule"))
-				assert.Equal(t, len(scds.Placement.TopologySpreadConstraints), 2)
-				assert.Equal(t, scds.Placement.TopologySpreadConstraints[0].TopologyKey, defaults.RackTopologyKey)
+				assert.Equal(t, scds.Placement.TopologySpreadConstraints[0].WhenUnsatisfiable, corev1.UnsatisfiableConstraintAction("DoNotSchedule"), c.label)
+				assert.Equal(t, len(scds.Placement.TopologySpreadConstraints), 2, c.label)
+				assert.Equal(t, scds.Placement.TopologySpreadConstraints[0].TopologyKey, defaults.RackTopologyKey, c.label)
 				placementOsd := getPlacement(c.sc, "osd-tsc")
 				newTSC := placementOsd.TopologySpreadConstraints[0]
 				newTSC.TopologyKey = defaults.RackTopologyKey
 				newTSC.WhenUnsatisfiable = corev1.UnsatisfiableConstraintAction("DoNotSchedule")
 				placementOsd.TopologySpreadConstraints = []corev1.TopologySpreadConstraint{newTSC, placementOsd.TopologySpreadConstraints[0]}
-				assert.Equal(t, placementOsd, scds.Placement)
+				assert.Equal(t, placementOsd, scds.Placement, c.label)
 			} else {
-				assert.Equal(t, getPlacement(c.sc, "osd-tsc"), scds.Placement)
+				assert.Equal(t, getPlacement(c.sc, "osd-tsc"), scds.Placement, c.label)
 			}
 			if c.lenOfMatchExpression == 0 {
-				assert.Nil(t, scds.Placement.NodeAffinity)
+				assert.Nil(t, scds.Placement.NodeAffinity, c.label)
 			} else {
 				matchExpressions := scds.Placement.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions
-				assert.Equal(t, c.lenOfMatchExpression, len(matchExpressions))
+				assert.Equal(t, c.lenOfMatchExpression, len(matchExpressions), c.label)
 			}
 			topologyKey := scds.PreparePlacement.TopologySpreadConstraints[0].TopologyKey
 
 			if c.topologyKey == "rack" {
-				assert.Equal(t, defaults.RackTopologyKey, topologyKey)
-			} else if len(c.sc.Status.NodeTopologies.Labels[zoneTopologyLabel]) >= defaults.DeviceSetReplica {
-				assert.Equal(t, zoneTopologyLabel, topologyKey)
+				assert.Equal(t, defaults.RackTopologyKey, topologyKey, c.label)
+			} else if len(c.sc.Status.NodeTopologies.Labels[zoneTopologyLabel]) >= (deviceSet.Count / getReplicasPerFailureDomain(c.sc)) {
```

Comment:
This change should be included in the same commit as the code changes, since it is technically required to allow the code commit to pass unit tests.